### PR TITLE
Bug 1149878 - [MCTS][2.2] Enhance Device Profile to load previous save

### PR DESCRIPTION
**Pull request by oouyang**
1. enhance profile loaded to profile.html
2. fix empty parameter not pass profile parameter to subsuite
3. pass profile loaded from file to tonado server page(profile.html)
4. fix tonado server cannot open in some platform
5. fix typo of unlink device_profile file
6. add tabzilla.png instead using base64 embedded in profile.html
7. profile.html javascript to process message from python and display in UI

### CODE REVIEW

File: certsuite/harness.py [hunk 1]
```diff
@@ -382,19 +382,19 @@ def list_tests(args, config):
...
         environment.env.handler = server.wait_for_client()
-        
-        question = ''
+
         resp = environment.env.handler.prompt(question)
 
-        message = 'Create device profile failed!!'    
-        if resp != '':
+        message = 'Create device profile failed!!'
+        if resp != '' and resp != None:
```
Comment (altsai) [inline certsuite/harness.py]: for the better way to use python, you should use
```python
if resp is not None and resp != '':
```
It brings better performance and easier for process.

File: certsuite/harness.py [hunk 2]
```diff
@@ -421,6 +422,63 @@ def check_device_profile(device_profile_path):
                     logger.error('Invalide device profile file [%s]' % device_profile_path)
     except:
         logger.critical("Encountered error at checking device profile file [%s]:\n%s" % (device_profile_path, traceback.format_exc()))
+    finally:
+        return device_profile_object
+
+def prepare_device_profile(edit_profile_path, profile_path, suites):
+    profile_object = { 'return' : 'cancel' }
```
Comment (altsai) [inline certsuite/harness.py]: PEP8, should be:
```python
    profile_object = {'return': 'cancel'}
```

File: certsuite/harness.py [hunk 3]
```diff
@@ -382,19 +380,19 @@ def list_tests(args, config):
...
-        
-        question = ''
-        resp = environment.env.handler.prompt(question)
 
-        message = 'Create device profile failed!!'    
-        if resp != '':
+        resp = environment.env.handler.prompt(message)
+
+        message = 'Create device profile failed!!'
+        if resp is not None and resp != '':
```
Comment (altsai) [inline certsuite/harness.py]: err... That would be my fault if I didn't make it clear. I apologize for asking refine twice. None and empty string are all falsy statement. Therefore, the following code should work:
```python
if not resp:
```

File: certsuite/harness.py [hunk 4]
```diff
@@ -421,6 +420,49 @@ def check_device_profile(device_profile_path):
                     logger.error('Invalide device profile file [%s]' % device_profile_path)
```
Comment (altsai) [inline certsuite/harness.py]: just catch that you can merge
'''python
else:
    if not 'contact' in device_profile_object['result']:
'''
to
'''python
elif not 'contact' in device_profile_object['result']
'''

File: certsuite/harness.py [hunk 5]
```diff
@@ -421,6 +420,49 @@ def check_device_profile(device_profile_path):
...
     except:
         logger.critical("Encountered error at checking device profile file [%s]:\n%s" % (device_profile_path, traceback.format_exc()))
+    finally:
+        return device_profile_object
```
Comment (altsai) [inline certsuite/harness.py]: I wonder would that cause any trouble if there's a valid json file without 'result' in it would return to caller.

Comment (oouyang) [inline certsuite/harness.py]: If the profile is invalid in these two condition, I will use edit_profile to recover it in coming fix

Comment (altsai) [inline certsuite/harness.py]: got it.

**Pull request merged**